### DISCUSSION TITLE
Projection Matrix, View Matrix and Camera Matrix confused with the terms!

### DISCUSSION THREAD

**Discussion by aprath1** [bulletphysics/bullet3#3970] [Q&A]
What exactly does computeViewMatrix and computeProjectionMatrixFOV give in pybullet?
Is it like this:
1. computeViewMatrix = camera extrinsic matrix 
2. computeProjectionMatrixFOV = camera intrinsic matrix 
Then the camera matrix can be computed from 1 and 2.

OR

is it like the computeProjectionMatrixFOV is the projection matrix of the camera matrix itself?
The terms are confusing to me, since some online materials use the term [Projection Matrix](http://www5.cs.fau.de/conrad/doc/index3286.html?edu/stanford/rsl/conrad/geometry/Projection.html) and [Camera Matrix ](http://ksimek.github.io/2012/08/14/decompose/) interchangeably. Both terms being defined as a matrix which transforms a world coordinate point into into a image pixel coordinate.

What I finally need to understand is given pybullet ViewMatrix and ProjectionMatrix **how to ?**  
1. get the camera intrinsic and extrinsic matrix
2. get the camera matrix or projection matrix (a matrix which can transform a world coordinate point to image pixel coordinate.

**Comment by erwincoumans** [comment]
the view and projection matrix are standard in OpenGL, you may find more information in OpenGL related forums.
